### PR TITLE
Fix pkg-config support on Nix

### LINKED ISSUE #2815: Wrong usage of GNUInstallDirs when generating pkg-config files
### Prerequisite Checklist

- [X] I searched for [existing issues](https://github.com/search?q=repo%3ASFML%2FSFML&type=issues) to prevent duplicates
- [X] I searched for [existing discussions on the forum](https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fen.sfml-dev.org) to prevent duplicates
- [X] I am here to report an issue and not to just ask a question or look for help (use the [forum](https://en.sfml-dev.org/forums/index.php#c3) or [Discord](https://discord.gg/nr4X7Fh) instead)

### Describe your issue here

SFML uses GNUInstallDirs variables (such as CMAKE_INSTALL_LIBDIR) to generate pkg-config files (located in tools/pkg-config).
The problem is that variables like CMAKE_INSTALL_LIBDIR aren't guaranteed to be a path relative to CMAKE_INSTALL_PREFIX, they can even be an absolute path!
This is described in the [official documentation](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html), where it says "However, an absolute path is also allowed" under CMAKE_INSTALL_\<dir>.

All files in tools/pkg-config/*.pc.in define libdir as `${exec_prefix}/@CMAKE_INSTALL_LIBDIR@`, which might concatenate two absolute paths together. This is the case on NixOS for example.

This problem should be easy to fix using cmake_path() as described below:
https://github.com/jtojnar/cmake-snips#concatenating-paths-when-building-pkg-config-files

Also, the install location of *.pc files is affected. Check: [this line in SFML 2.6.x](https://github.com/SFML/SFML/blob/c1c65b53ec2c13e25c83fed5fee9c22562736a60/CMakeLists.txt#L306C15-L306C15) and also [this line in SFML master](https://github.com/SFML/SFML/blob/4e2f6ae655e224b67e4721b99fd52d7f16f49442/cmake/Config.cmake#L69C9-L69C9).
Both define SFML_PKGCONFIG_DIR variable using CMAKE_INSTALL_LIBDIR which later gets concatenated with CMAKE_INSTALL_PREFIX.

### Your Environment

- OS / distro / window manager: Linux / NixOS / XMonad
- SFML version: 2.6.1
- Compiler / toolchain: g++ / gcc
- Special compiler / CMake flags: -DSFML_INSTALL_PKGCONFIG_FILES=TRUE -DSFML_USE_SYSTEM_DEPS=TRUE

### Steps to reproduce

If it helps, I'm packaging SFML 2.6.1 on NixOS [here](https://github.com/ThwyIgo/cmake-sfml-bloat-project/blob/1d62e3ad110bf95e3489c1599e715bd27b63adfa/nix/sfml.nix).
I made a temporary workaround to fix *.pc.in files using sed, but the install location is still wrong.

You can build the derivation and check where pkg-config files were installed. In my system, they're at
/nix/store/\<hash>-sfml-2.6.1/nix/store/\<hash>-sfml-2.6.1/lib/pkgconfig
Clearly two absolute paths were concatenated.

### Expected behavior

pkg-config files installed in /nix/store/\<hash>-sfml-2.6.1/lib/pkgconfig
libdir defined as ${exec_prefix}/lib

### Actual behavior

pkg-config files installed in /nix/store/\<hash>-sfml-2.6.1/nix/store/\<hash>-sfml-2.6.1/lib/pkgconfig
libdir defined as ${exec_prefix}/\<CMAKE_INSTALL_PREFIX>/lib

### PR DESCRIPTION
**Pull request by ChrisThrasher**
## Description

Closes #2815. I also added a missing changelog entry that we meant to add earlier.

### CODE REVIEW

**Comment by ChrisThrasher** [comment]
@dkosmari You've previously helped debug pkg-config support. Do you mind testing this to make sure it hasn't broken your workflows?

**Comment by ChrisThrasher** [comment]
@ThwyIgo Please check this out and approve the PR if it looks good to you.

**Comment by ThwyIgo** [comment]
> @ThwyIgo Please check this out and approve the PR if it looks good to you.

LGTM, I approve. But as a random person who doesn't even have a profile picture, I don't have permission to "approve" the PR.

BTW, I'm not the pkg-config guy, but I tested it compiling a simple project I have here with the command:
`g++ src/*.cpp $(pkg-config --cflags --libs sfml-all) -lboost_filesystem -lboost_system -Isrc/include`
Success. (boost has nothing to do with SFML, it's just my project dependency).
Compiling with CMake also works.

The output of `pkg-config --cflags --libs sfml-all` on my system (NixOS) is:
`-I/nix/store/aiy5hvk2ha3l24qx2rssf00zxklq00q9-sfml-2.6.1/include -L/nix/store/aiy5hvk2ha3l24qx2rssf00zxklq00q9-sfml-2.6.1/lib -lsfml-graphics -lsfml-window -lsfml-audio -lsfml-network -lsfml-system`
Which is correct for NixOS.

**Comment by ThwyIgo** [comment]
Oh, actually, I have a suggestion not directly related to the bug.

Why are we using GNUInstallDirs to define libdir in pkg-config, but not for includedir?
There's CMAKE_INSTALL_INCLUDEDIR which we can convert to a relative path and use it in pkg-config.

This would be great, because it would be easy to install .so files in one place, and .hpp files in another place. That's what some libraries do in NixOS. Usually the derivation (package) is divided in "dev" and "out", dev contains .hpp and out contains .so. This allows users of the distro to save space by not having .hpp files in their system.

**Comment by dkosmari** [comment]
> @dkosmari You've previously helped debug pkg-config support. Do you mind testing this to make sure it hasn't broken your workflows?

Tested on Fedora 39, the pkg-config paths are still fine, on both native and cross mingw builds.

File: changelog.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 -   Ensure GNUInstallDirs cache vars are included before first used (#2778, #2779)
 -   [macOS] Fix incorrect variable expansion (#2780)
+-   Issue warning when trying to use UCRT MinGW with precompiled MSVCRT depenencies (#2821)
```
Comment (eXpl0it3r) [inline changelog.md:9]: <3

**Review by eXpl0it3r — APPROVED** [review]

**Pull request merged**